middleware_test.go

```diff
@@ -24,24 +25,14 @@ func TestMiddlewareAdd(t *testing.T) {
 	router := NewRouter()
 	router.HandleFunc("/", dummyHandler).Methods("GET")
 
-	mw := &testMiddleware{}
-
-	router.useInterface(mw)
-	if len(router.middlewares) != 1 || router.middlewares[0] != mw {
-		t.Fatal("Middleware was not added correctly")
-	}
-
-	router.Use(mw.Middleware)
-	if len(router.middlewares) != 2 {
-		t.Fatal("MiddlewareFunc method was not added correctly")
-	}
-
-	banalMw := func(handler http.Handler) http.Handler {
-		return handler
-	}
-	router.Use(banalMw)
-	if len(router.middlewares) != 3 {
-		t.Fatal("MiddlewareFunc method was not added correctly")
+	for i := 0; i < 3; i++ {
+		t.Run(fmt.Sprintf("adds %d middlewares", i+1), func(t *testing.T) {
```

Comment:
Describe the test better. Not sure a sub-test makes as much sense here either?

Reply:
Yeah, this one felt weird. I'll revert changing this to subtests. :+1: